dll/win32/shell32/dialogs/drive.cpp

```diff
@@ -24,14 +24,26 @@
 
 WINE_DEFAULT_DEBUG_CHANNEL(shell);
 
+EXTERN_C NTSTATUS WINAPI NtQueryObject(HANDLE, OBJECT_INFORMATION_CLASS, PVOID, ULONG, PULONG);
+
 typedef struct
 {
-    WCHAR   Drive;
+    WCHAR   Drive; // Note: This is the drive number, not a drive letter
     UINT    Options;
     UINT Result;
     BOOL bFormattingNow;
+    HWND hWndMain;
+    HWND hWndTip, hWndTipTrigger;
+    WCHAR szTip[400];
```

Comment:
Please change this to
```suggestion
    struct { UNICODE_STRING Name; WCHAR Buffer[MAX_PATH]; } Tip;
```
(changing the "MAX_PATH" to whatever size suits you), in order to avoid the ugly casts you need to do later in the GetTipText() function.
(NOTE: The `UNICODE_STRING Name;` is to avoid using `OBJECT_NAME_INFORMATION;` that would require including the `<ndk/obtypes.h>` , or <ndk/obfuncs.h>` if you want to also have the NtQueryObject definition for free.)